lib/std/time.zig

```diff
@@ -50,51 +52,45 @@ pub fn sleep(nanoseconds: u64) void {
     std.os.nanosleep(s, ns);
 }
 
+test "sleep" {
+    sleep(1);
+}
+
 /// Get a calendar timestamp, in seconds, relative to UTC 1970-01-01.
 /// Precision of timing depends on the hardware and operating system.
-/// The return value is signed because it is possible to have a date that is
-/// before the epoch.
 /// See `std.os.clock_gettime` for a POSIX timestamp.
-pub fn timestamp() i64 {
+pub fn timestamp() u64 {
     return @divFloor(milliTimestamp(), ms_per_s);
 }
 
 /// Get a calendar timestamp, in milliseconds, relative to UTC 1970-01-01.
 /// Precision of timing depends on the hardware and operating system.
-/// The return value is signed because it is possible to have a date that is
-/// before the epoch.
 /// See `std.os.clock_gettime` for a POSIX timestamp.
-pub fn milliTimestamp() i64 {
-    return @intCast(i64, @divFloor(nanoTimestamp(), ns_per_ms));
+pub fn milliTimestamp() u64 {
+    return @divFloor(nanoTimestamp(), ns_per_ms);
 }
 
 /// Get a calendar timestamp, in nanoseconds, relative to UTC 1970-01-01.
 /// Precision of timing depends on the hardware and operating system.
 /// On Windows this has a maximum granularity of 100 nanoseconds.
-/// The return value is signed because it is possible to have a date that is
-/// before the epoch.
 /// See `std.os.clock_gettime` for a POSIX timestamp.
-pub fn nanoTimestamp() i128 {
-    if (is_windows) {
-        // FileTime has a granularity of 100 nanoseconds and uses the NTFS/Windows epoch,
-        // which is 1601-01-01.
-        const epoch_adj = epoch.windows * (ns_per_s / 100);
-        var ft: os.windows.FILETIME = undefined;
-        os.windows.kernel32.GetSystemTimeAsFileTime(&ft);
-        const ft64 = (@as(u64, ft.dwHighDateTime) << 32) | ft.dwLowDateTime;
-        return @as(i128, @bitCast(i64, ft64) + epoch_adj) * 100;
-    }
-    if (builtin.os.tag == .wasi and !builtin.link_libc) {
-        var ns: os.wasi.timestamp_t = undefined;
-        const err = os.wasi.clock_time_get(os.wasi.CLOCK_REALTIME, 1, &ns);
-        assert(err == .SUCCESS);
-        return ns;
-    }
-    var ts: os.timespec = undefined;
-    os.clock_gettime(os.CLOCK_REALTIME, &ts) catch |err| switch (err) {
-        error.UnsupportedClock, error.Unexpected => return 0, // "Precision of timing depends on hardware and OS".
+pub fn nanoTimestamp() u64 {
```

Comment:
Why 64 bits instead of 128 bits?

Reply:
Was for consistency with Timer. Thinking about it again, `Timer.read()` returns elapsed time & program wouldn't be running for 500+ years to overflow so they shouldn't be compared.